src/main/java/org/spongepowered/api/Server.java

```diff
@@ -429,4 +429,13 @@
      */
     void setPlayerIdleTimeout(int timeout);
 
+    /**
+     * Tests if the {@link Thread#currentThread current thread} is the
+     * main thread of the server.
+     *
+     * @return {@code true} if current thread is the main
+     *     thread, {@code false} if not
+     */
+    boolean isMainThread();
```

Comment:
@kashike 

Something like what @simon816 is proposing. When someone reads "main thread", that inherently implies only one and should client API ever to be added, the Client interface would have a method just like this one.